### PR TITLE
[FIX] Fixes concurrency issues for LogStream on Mac and avoids access…

### PR DESCRIPTION
**Pull request by jpfeuffer**
…ing omp symbols in dtor in case of premature closing.

### CODE REVIEW

File: src/openms/source/CONCEPT/LogStream.cpp [hunk 1]
```diff
@@ -425,8 +427,9 @@ namespace OpenMS
     void LogStreamNotifier::registerAt(LogStream & log)
     {
       unregister();
-
-      registered_at_ = &log;
+      {
```
Comment (timosachsenberg) [inline src/openms/source/CONCEPT/LogStream.cpp]: why the brackets?

Comment (jpfeuffer) [reply src/openms/source/CONCEPT/LogStream.cpp]: I probably had a pragma there before.

**Comment by hroest** [comment]
rebuild jenkins

File: src/openms/source/CONCEPT/LogStream.cpp [hunk 2]
```diff
@@ -80,10 +80,7 @@ namespace OpenMS
 
     LogStreamBuf::~LogStreamBuf()
     {
-      sync();
-#ifdef _OPENMP
-#pragma omp critical
-#endif
+      syncLF_();
```
Comment (hroest) [inline src/openms/source/CONCEPT/LogStream.cpp]:
```suggestion
      // Prevent issue on OSX with OpenMP: destructors of global objects seem to be called after tearing down the OpenMP context, we therefore cannot use any locks here.
      syncLF_();
```

File: src/openms/source/CONCEPT/LogStream.cpp (hunk 1 above)
Comment (hroest) [inline src/openms/source/CONCEPT/LogStream.cpp]:
```suggestion
```

File: src/openms/source/CONCEPT/LogStream.cpp [hunk 3]
```diff
@@ -425,8 +427,9 @@ namespace OpenMS
     void LogStreamNotifier::registerAt(LogStream & log)
     {
       unregister();
-
-      registered_at_ = &log;
+      {
+        registered_at_ = &log;
+      };
```
Comment (hroest) [inline src/openms/source/CONCEPT/LogStream.cpp]:
```suggestion
```

**Comment by hroest** [comment]
looks, good -- would be even better if the tests on OSX would run to test whether it works

**Comment by jpfeuffer** [comment]
Thanks! Will adapt the changes. Jenkins seemed to be  down on the weekend.

**Comment by jpfeuffer** [comment]
rebuild jenkins

**Comment by timosachsenberg** [comment]
Looks good to me

**Pull request merged**